src/checksum.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+/// Stateful checksum calculator.
+///
+/// Ext4 has [metadata checksums][0] for most data structures
+/// (superblock, block group descriptor, inode, etc).
+///
+/// [0]: https://www.kernel.org/doc/html/latest/filesystems/ext4/overview.html#checksums
+///
+/// # Seed
+///
+/// The default seed is `0xffff_ffff`. This is used for the superblock
+/// checksum. Other structures use a checksum seed stored in the
+/// superblock. The checksum seed is derived from the filesystem's
+/// initial UUID.
+#[derive(Clone)]
+pub(crate) struct Checksum {
+    digest: crc::Digest<'static, u32>,
+}
+
+impl Checksum {
+    /// Create a `Checksum` with the default seed.
+    pub(crate) fn new() -> Self {
+        Self::with_seed(0xffff_ffff)
+    }
+
+    /// Create a `Checksum` with the given `seed`.
+    pub(crate) fn with_seed(seed: u32) -> Self {
+        const CRC32C: crc::Crc<u32> = crc::Crc::<u32>::new(&crc::CRC_32_ISCSI);
```

Comment:
What's ISCSI got to do with it?

Reply:
Just one of many names for this algorithm: https://reveng.sourceforge.io/crc-catalogue/all.htm#crc.cat.crc-32-iscsi

I think CRC32C and Castagnoli are the more common names.